src/components/Executions/useChildNodeExecutions.ts

```diff
@@ -1,91 +1,100 @@
-    apiContext,
+    dataCache,
     config,
     taskExecutionId
 }: FetchGroupForTaskExecutionArgs): Promise<NodeExecutionGroup> {
     return {
         // NodeExecutions created by a TaskExecution are grouped
         // by the retry attempt of the task.
         name: formatRetryAttempt(taskExecutionId.retryAttempt),
-        nodeExecutions: await fetchTaskExecutionChildren(
-            { config, taskExecutionId },
-            apiContext
+        nodeExecutions: await dataCache.getTaskExecutionChildren(
+            taskExecutionId,
+            config
         )
     };
 }
 
 interface FetchGroupForWorkflowExecutionArgs {
-    apiContext: APIContextValue;
     config: RequestConfig;
+    dataCache: ExecutionDataCache;
     workflowExecutionId: WorkflowExecutionIdentifier;
 }
 async function fetchGroupForWorkflowExecution({
-    apiContext,
     config,
+    dataCache,
     workflowExecutionId
 }: FetchGroupForWorkflowExecutionArgs): Promise<NodeExecutionGroup> {
     return {
         // NodeExecutions created by a workflow execution are grouped
         // by the execution id, since workflow executions are not retryable.
         name: workflowExecutionId.name,
-        nodeExecutions: await fetchNodeExecutions(
-            { config, id: workflowExecutionId },
-            apiContext
+        nodeExecutions: await dataCache.getNodeExecutions(
+            workflowExecutionId,
+            config
         )
     };
 }
 
 interface FetchNodeExecutionGroupArgs {
-    apiContext: APIContextValue;
     config: RequestConfig;
+    dataCache: ExecutionDataCache;
     nodeExecution: NodeExecution;
 }
 
 async function fetchGroupsForTaskExecutionNode({
-    apiContext,
     config,
+    dataCache,
     nodeExecution: { id: nodeExecutionId }
 }: FetchNodeExecutionGroupArgs): Promise<NodeExecutionGroup[]> {
-    const taskExecutions = await fetchTaskExecutions(
-        nodeExecutionId,
-        apiContext
-    );
+    const taskExecutions = await dataCache.getTaskExecutions(nodeExecutionId);
 
-    return await Promise.all(
-        taskExecutions.map(({ id: taskExecutionId }) =>
-            fetchGroupForTaskExecution({ apiContext, config, taskExecutionId })
+    // For TaskExecutions marked as parents, fetch its children and create a group.
+    // Otherwise, return null and we will filter it out later.
+    const groups = await Promise.all(
+        taskExecutions.map(execution =>
+            execution.isParent
+                ? fetchGroupForTaskExecution({
+                      dataCache,
+                      config,
+                      taskExecutionId: execution.id
+                  })
+                : Promise.resolve(null)
         )
     );
+
+    // Remove any empty groups
+    return groups.reduce<NodeExecutionGroup[]>((out, group) => {
+        if (group === null || group.nodeExecutions.length === 0) {
+            return out;
+        }
+        return [...out, group];
+    }, []);
```

Comment:
Feels like a filter is a more applicable function to use here

```suggestion
    return groups.filter((group) => {
        return group !== null && group.nodeExecutions.length;
    }, []);
```

Reply:
Eek. I agree. I'm not sure why this is a reduce and not a filter. Let me refactor it.